packages/ra-tree-ui-materialui/src/Tree.tsx

```diff
@@ -0,0 +1,235 @@
+    aside?: ReactElement<any>;
+    children: ReactElement<any>;
+    className?: string;
+    exporter?: Exporter;
+    filter?: object;
+    parentSource: string;
+    positionSource?: string;
+    title?: string;
+}
+
+interface InjectedProps {
+    basePath: string;
+    resource: string;
+    version: number;
+}
+
+export class Tree extends Component<Props & InjectedProps> {
+    static propTypes = {
+        actions: PropTypes.element,
+        aside: PropTypes.element,
+        basePath: PropTypes.string.isRequired,
+        children: PropTypes.node,
+        classes: PropTypes.object,
+        className: PropTypes.string,
+        exporter: PropTypes.func,
+        filter: PropTypes.object,
+        parentSource: PropTypes.string.isRequired,
+        positionSource: PropTypes.string,
+        resource: PropTypes.string.isRequired,
+        title: PropTypes.string,
+        version: PropTypes.number,
+    };
+
+    static defaultProps = {
+        classes: {},
+        parentSource: 'parent_id',
+        positionSource: 'position',
+    };
+
+    componentDidMount() {
+        const childrenCount = Children.count(this.props.children);
+
+        if (childrenCount > 1 && process.env.NODE_ENV !== 'production') {
+            warnAboutChildren();
+        }
+    }
+
+    render() {
+        const { children, parentSource, positionSource, ...props } = this.props;
+        return (
+            <DragDropContextProvider backend={HTML5Backend}>
+                <TreeController
+                    parentSource={parentSource}
+                    positionSource={positionSource}
+                    {...props}
+                >
+                    {controllerProps => (
+                        <TreeView {...controllerProps}>{children}</TreeView>
+                    )}
+                </TreeController>
+            </DragDropContextProvider>
+        );
+    }
+}
+
+export default Tree as ComponentType<Props>;
+
+interface ViewProps {
+    defaultTitle: string;
+}
+
+export const styles = createStyles({
+    root: {
+        display: 'flex',
+        flex: 1,
+    },
+    card: {
+        position: 'relative',
+        flex: '1 1 auto',
+    },
+    actions: {
+        zIndex: 2,
+        display: 'flex',
+        justifyContent: 'flex-end',
+        flexWrap: 'wrap',
+    },
+});
+
+const TreeView = withStyles(styles)(
+    ({
+        actions = <TreeListActions />,
+        aside,
+        children,
+        className,
+        classes,
+        defaultTitle,
+        exporter,
+        filter,
+        title,
+        version,
+        ...props
+    }: Props & InjectedProps & ViewProps & WithStyles<typeof styles>) => (
+        <div
+            className={classnames('tree-page', classes.root, className)}
+            {...sanitizeRestProps(props)}
+        >
+            <Title title={title} defaultTitle={defaultTitle} />
+            <Card className={classes.card}>
+                {actions && (
+                    <TreeListToolbar
```

Comment:
how is that different from a `ListToolbar`?

Reply:
Filters